scripts/language_model/word_language_model.py

```diff
@@ -216,12 +310,16 @@ def evaluate(data_source, batch_size, ctx=None):
     """
     total_L = 0.0
     ntotal = 0
-    hidden = model.begin_state(batch_size, func=mx.nd.zeros, ctx=context[0])
+    if segment == 'val':
+        model_eval.load_params(args.save + '.val', context)
```

Comment:
using the trick mentioned above, we should be able to avoid loading different parameters.

Reply:
This logic has nothing to do with the trick. We need to update the model_eval parameters before generate the val ppl, but only update the parameters of the model once the val ppl < best ppl.

Reply:
if it's the same instance, then there's no need to save and then load parameter. or did I miss something?